rpc/src/v1/impls/ethcore.rs

```diff
@@ -334,4 +334,17 @@ impl<C, M, S: ?Sized, F> Ethcore for EthcoreClient<C, M, S, F> where
 		self.dapps_port
 			.ok_or_else(|| errors::dapps_disabled())
 	}
+
+	fn next_nonce(&self, address: H160) -> Result<U256, Error> {
+		try!(self.active());
+		let address: Address = address.into();
+		let miner = take_weak!(self.miner);
+		let client = take_weak!(self.client);
+
+		Ok(miner.last_nonce(&address)
+			.map(|n| n + 1.into())
+			.unwrap_or_else(|| client.latest_nonce(&address))
```

Comment:
`latest_nonce` in client is actually returning the a nonce for next transaction (i.e. it returns `0x0` when there are no transactions yet)
`miner.last_nonce` returns last nonce of transactions transaction queue so we need to add one before returning.